### PR TITLE
fix: preserve total's sign in distributeBy rest correction

### PR DESCRIPTION
**Pull request by andersem**
- `Money.distributeBy` could return a part with the **sign opposite to the total** when the rounded proportional shares overshot and a part rounded to `0.00` happened to be the first weighted index the rest-correction loop visited.
- Skip positions where the smallest-unit adjustment would flip a part's sign relative to the total. By the rounding-bound argument the capacity always exists in same-sign parts, so the loop still terminates.
- Add regression tests for both positive and negative totals.

## Reproducer

```ts
Money.fromFractionlessAmount(269, \"NOK\")
    .distributeBy([100, 100, 57900, 9400, 74900, 9400])
// before: ['-0.01', '0.00', '1.03', '0.17', '1.33', '0.17']
// after:  ['0.00',  '0.00', '1.02', '0.17', '1.33', '0.17']
```

Both versions sum to 2.69 NOK (the invariant `sum(parts) === this` is preserved either way) — but the \"before\" output violates the implicit invariant that callers rely on: parts share the sign of the total. Returning a negative part for a positive total made the result unusable as a weight in a subsequent `distributeBy` call.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -522,11 +522,37 @@ export class Money {
...
+                    rest = rest.subtract(smallestUnit);
+                    skipsInARow = 0;
+                } else {
+                    skipsInARow++;
+                    // Defensive bound: capacity analysis says this is
+                    // unreachable, but don't risk an infinite loop on a
+                    // pathological input.
+                    if (skipsInARow >= weights.length) {
+                        break;
+                    }
```
[the remark this answers is not in the document]
Comment (andersem) [reply src/index.ts:558]: Fixed in b535e8b. Replaced the silent break with a throw so a non-zero rest can't escape as a sum(parts) !== this violation.

**Review by frodeaa — APPROVED** [review]
